ignite/pkg/cosmosanalysis/app/app_test.go

```diff
@@ -246,6 +267,17 @@ func TestCheckKeeper(t *testing.T) {
 	err = app.CheckKeeper(tmpDir, "BarKeeper")
 	require.Error(t, err)
 
+	// Test with a source file containing an app with generics
+	tmpDirGenericApp := t.TempDir()
+	tmpFileGenericApp := filepath.Join(tmpDirGenericApp, "app.go")
+	err = os.WriteFile(tmpFileGenericApp, GenericAppFile, 0644)
+	require.NoError(t, err)
+
+	err = app.CheckKeeper(tmpDirGenericApp, "FooKeeper")
+	require.NoError(t, err)
+	err = app.CheckKeeper(tmpDirGenericApp, "BarKeeper")
+	require.Error(t, err)
```

Comment:
Adding the test in a separate `TestCheckKeeperWithGenerics` function might be a good idea, it will also make the test variable names cleaner by being able to use `tmpDir` instead of `tmpDirGenericApp`, or similarly with `tmpFileGenericApp`.

Actually the previously existing `TestCheckKeeper` could be splitted into three different tests, and maybe the files should have been real files in a `testdata` folder instead of variables.

Reply:
@jeronimoalbi done f59d38b8293f680b29d42c000e18d96b5e61941c

- I've used a parameterized test instead of splitting into multiple methods
- I moved the files content into real files in `testdata` and import them in the test using `embed`.